projects/gloo/pkg/translator/listener.go

```diff
@@ -184,30 +182,19 @@ func mergeSslConfigs(sslConfigs []*v1.SslConfig) []*v1.SslConfig {
 
 	for _, sslConfig := range sslConfigs {
 
-		key := ""
-		switch sslCfg := sslConfig.SslSecrets.(type) {
-		case *v1.SslConfig_SecretRef:
-			key = sslCfg.SecretRef.GetName() + "," + sslCfg.SecretRef.GetNamespace()
-		case *v1.SslConfig_SslFiles:
-			key = sslCfg.SslFiles.GetTlsCert() + "," + sslCfg.SslFiles.GetTlsKey() + "," + sslCfg.SslFiles.GetRootCa()
-		default:
-			result = append(result, sslConfig)
-			continue
-		}
+		// make sure ssl configs are only different by sni domains
+		sslConfigCopy := *sslConfig
+		sslConfigCopy.SniDomains = nil
+		hash, _ := sslConfigCopy.Hash(nil)
 
-		tmp := make([]string, len(sslConfig.VerifySubjectAltName))
-		copy(tmp, sslConfig.VerifySubjectAltName)
-		sort.Strings(tmp)
-		key = key + ";" + strings.Join(tmp, ",")
+		key := fmt.Sprintf(";%d", hash)
 
 		if matchingCfg, ok := mergedSslSecrets[key]; ok {
-			matchingCfg.SslSecrets = sslConfig.SslSecrets
-			matchingCfg.VerifySubjectAltName = sslConfig.VerifySubjectAltName
 			if len(matchingCfg.SniDomains) == 0 || len(sslConfig.SniDomains) == 0 {
 				// if either of the configs match on everything; then match on everything
 				matchingCfg.SniDomains = nil
 			} else {
-				matchingCfg.SniDomains = append(mergedSslSecrets[key].SniDomains, sslConfig.SniDomains...)
+				matchingCfg.SniDomains = merge(mergedSslSecrets[key].SniDomains, sslConfig.SniDomains...)
```

Comment:
This line is a bit confusing because the same value is referred to as `matchingCfg.SniDomains` and `mergedSslSecrets[key].SniDomains`.
```suggestion
				matchingCfg.SniDomains = merge(matchingCfg.SniDomains, sslConfig.SniDomains...)
```